SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -2173,3 +2258,189 @@ void USpatialReceiver::PeriodicallyProcessIncomingRPCs()
 		}
 	}, GetDefault<USpatialGDKSettings>()->QueuedIncomingRPCWaitTime, true);
 }
+
+bool USpatialReceiver::NeedToLoadClass(const FString& ClassPath)
+{
+	return FindObject<UClass>(nullptr, *ClassPath, false) == nullptr;
+}
+
+FString USpatialReceiver::GetPackagePath(const FString& ClassPath)
+{
+	return FSoftObjectPath(ClassPath).GetLongPackageName();
+}
+
+void USpatialReceiver::OnAsyncPackageLoaded(const FName& PackageName, UPackage* Package, EAsyncLoadingResult::Type Result)
+{
+	FString PackageNameStr = PackageName.ToString();
+	if (!AsyncLoadingPackages.Contains(PackageNameStr))
+	{
+		UE_LOG(LogSpatialReceiver, Error, TEXT("USpatialReceiver::OnAsyncPackageLoaded: Package loaded but no entry in AsyncLoadingPackages. Package: %s"), *PackageNameStr);
+		return;
+	}
+
+	TArray<Worker_EntityId> Entities = AsyncLoadingPackages.FindAndRemoveChecked(PackageNameStr);
+
+	for (Worker_EntityId Entity : Entities)
+	{
+		if (IsEntityWaitingForAsyncLoad(Entity))
+		{
+			UE_LOG(LogSpatialReceiver, Log, TEXT("Finished async loading package %s for entity %lld."), *PackageNameStr, Entity);
+
+			CriticalSectionSaveState CriticalSectionState;
+			if (bInCriticalSection)
+			{
+				SaveCriticalSection(CriticalSectionState);
+			}
+			bool bInCriticalSectionSaved = bInCriticalSection;
+			bInCriticalSection = true;
```

Comment:
Could this be stored into CriticalSectionSaveState's contructor instead ?

Reply:
It could, but I would rather keep CriticalSectionSaveState a plain struct.